### DISCUSSION TITLE
chat invite link error

### DISCUSSION THREAD

**Discussion by JonasCaetanoSz** [eternnoir/pyTelegramBotAPI#1945] [General]
hi, i'm creating a a bot that manages channel members , I save the id of the channels and groups that the bot was placed as an administrator in the Database, but when I use these id like this:

`for chat_id in self.database.get_all_chat():
      invite_url = self.bot.create_chat_invite_link(chat_id=chat_id).invite_link` 

I get the error: **bot is not a member of the supergroup chat**

I already checked, and all the channels and groups that are in the database the bot is admin, this seems to happen only with some chats, can you help me?

**Comment by coder2020official** [comment]
The id is mb outdated

**Reply by JonasCaetanoSz** [reply]
outdated chat_id? how do i update this? 

thanks for your answer (:

**Reply by coder2020official** [reply]
First, when u create a group, it is just a group, which has a random negative id
But as u do some actions, it is converted to supergroup. Supergroup's id will have -100 at the beginning.
So you are probably using group id instead of supergroup id

**Comment by Badiboy** [comment]
> bot is not a member of the supergroup chat

Come on, what is unclear in this message?! You can create invite link only when your bot is a member and an admin of the group with appropriate permissions. If it is not (currently - not a member).

**Reply by JonasCaetanoSz** [reply]
What's not clear about this message, is this serious? Because to me, it sounded like a joke. If I'm an administrator of the group or channel, obviously I'm already a member, or in my opinion, could be like that. This Telegram logic makes no sense. Anyway, I'm already correcting this issue using @coder2020official's tip.

**Comment by Badiboy** [comment]
![image](https://user-images.githubusercontent.com/5613421/225391135-9f2fc9b0-3a67-43cc-bb47-8d0edd81947f.png)

**Reply by JonasCaetanoSz** [reply]
Yes, I had already understood! Thank you for your help, both of you.  (:

I will be closing this as resolved.
